backend/src/helpers/databasefunctions.ts

```diff
@@ -16,31 +17,156 @@ const pool = new Pool(pgconfig);
 
 pool.on(
   'error',
-  async (_err: Error): Promise<never> => {
-    throw new ErrorHandler(400, { status: 'Database Error' });
+  async (err: Error): Promise<void> => {
+    // Uncertain how to properly solve this problem.
+    await parsePgError((err as unknown) as PgError);
   },
 );
 
+const hasConnection = async () => {
+  try {
+    await (await pool.connect()).query('SELECT NOW()');
+  } catch (e) {
+    await parsePgError(e);
+  }
+};
+
+const getClient = async (connection: Pool | Client): Promise<PoolClient | Client | undefined> => {
+  let client: any;
+  if (isPool(connection)) {
+    client = await connection.connect();
+    return client as PoolClient;
+  } else if (isClient(connection)) {
+    client = connection;
+    return client as Client;
+  } else {
+    client = undefined;
+    return client as undefined;
+  }
+};
+
+const disconnectClient = async (client: PoolClient | Client | undefined) => {
+  if (client === undefined) {
+    //TODO
+    throw new ErrorHandler(500, { type: 'Database Error' });
+  } else if (isPoolClient(client)) {
+    client.release();
+    return true;
+  } else if (isClient(client)) {
+    client.end();
+    return true;
+  }
+};
+
+const validateSQLStatement = async (sqlKeyword: string, sqlStatement: string): Promise<never | void> => {
+  const isValidSqlStatement = sqlStatement.toLowerCase().includes(sqlKeyword);
+  if (!isValidSqlStatement) {
+    throw new ErrorHandler(400, { type: 'Invalid SQL statment' });
+  }
+};
+
+const executeQuery = async (
+  sqlStatement: string,
+  connection: Pool | Client = pool,
+  data?: string[],
+): Promise<QueryResultRow | never> => {
+  let client: PoolClient | Client | undefined;
+  await hasConnection();
+  try {
+    client = await getClient(connection);
+    if (client !== undefined) {
+      const res = data ? await client.query(sqlStatement, data) : await client.query(sqlStatement);
+      return res;
+      //Consider throwing error, but it should be impossible for there not to be a connection.....
+    } else
+      throw new ErrorHandler(500, {
+        type: 'Logical Error, ask developers to review code.',
+      });
```

Comment:
There is no possibly way that the code should be able to reach that point, but i think TS threw me an error if i did not handle it this way. Because of that i placed an error there, just to check if i ever managed to get my code to reach that point.

Reply:
I removed it now, i have double checked, the code is not able to reach the else block.